notebooks/os_cnn_classification.ipynb

```diff
@@ -0,0 +1,291 @@
+       "    (averagepool): AdaptiveAvgPool1d(output_size=1)\n",
+       "    (hidden): Linear(in_features=40, out_features=50, bias=True)\n",
+       "  ),\n",
+       ")"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset_path = dirname(\"./notebooks/OS_CNN/UCRArchive_2018/\")\n",
+    "dataset_name = \"FiftyWords\"\n",
+    "\n",
+    "# load data,\n",
+    "X_train, y_train, X_test, y_test = TSC_data_loader(dataset_path, dataset_name)\n",
+    "print('train data shape', X_train.shape)\n",
+    "print()\n",
+    "print('train label shape',y_train.shape)\n",
+    "print('test data shape',X_test.shape)\n",
+    "print('test label shape',y_test.shape)\n",
+    "print('unique train label',np.unique(y_train))\n",
+    "print('unique test label',np.unique(y_test))\n",
+    "device = \"cpu\"\n",
+    "X_train, y_train, X_test, y_test = load_to_torch(X_train, y_train, X_test, y_test, device)\n",
+    "\n",
+    "# the model prints out the result every epoch\n",
+    "# defaul epoch size = 20\n",
+    "Max_kernel_size = 89\n",
+    "start_kernel_size = 1\n",
+    "# loss, optimizer, scheduler\n",
+    "input_channel = X_train.shape[1] # input channel size\n",
+    "n_class = max(y_train) + 1 # output class number\n",
+    "receptive_field_shape= min(int(X_train.shape[-1]/4),Max_kernel_size)\n",
+    "\n",
+    "model = OS_CNN_CLassifier(n_class.item(), input_channel, receptive_field_shape)\n",
+    "model.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "17a15659-94d5-424c-9d64-7fb3dd2ed3b7",
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "NameError",
+     "evalue": "name 'test_dataset' is not defined",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
+      "Input \u001b[0;32mIn [4]\u001b[0m, in \u001b[0;36m<cell line: 3>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m accuracy_list\u001b[38;5;241m=\u001b[39m[]\n\u001b[1;32m      2\u001b[0m correct \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[0;32m----> 3\u001b[0m N_test\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mlen\u001b[39m(\u001b[43mtest_dataset\u001b[49m)\n\u001b[1;32m      4\u001b[0m yhat \u001b[38;5;241m=\u001b[39m model\u001b[38;5;241m.\u001b[39mpredict(X_test)\n\u001b[1;32m      5\u001b[0m correct \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m (torch\u001b[38;5;241m.\u001b[39mtensor(yhat) \u001b[38;5;241m==\u001b[39m y_test)\u001b[38;5;241m.\u001b[39msum()\u001b[38;5;241m.\u001b[39mitem()\n",
+      "\u001b[0;31mNameError\u001b[0m: name 'test_dataset' is not defined"
+     ]
+    }
+   ],
+   "source": [
+    "accuracy_list=[]\n",
+    "correct = 0\n",
+    "N_test=len(test_dataset)\n",
```

Comment:
Shouldn't this code len(test_dataset) be X_test.shape[0] ?

Reply:
Yeah you are right! I've corrected that.